connector/interface.go

```diff
@@ -60,6 +60,11 @@ type ConnectorConfig interface {
 	Connector(ns url.URL, loginFunc oidc.LoginFunc, tpls *template.Template) (Connector, error)
 }
 
+// Groups is an optional interface implemented by a Connector.
+type Groups interface {
+	Groups(fullUserID string) ([]string, error)
```

Comment:
Do we expect this interface to only be implemented by connectors? Perhaps it would be best to embed the connector interface into this one to make that explicit.